include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_pressure_correction.cpp

```diff
@@ -340,83 +340,90 @@ TimeIntBDFPressureCorrection<dim, Number>::momentum_step()
   dealii::Timer timer;
   timer.restart();
 
-  // Extrapolate old solutions to get a good initial estimate for the solver.
-  if(this->use_extrapolation)
+  // in case we need to iteratively solve a linear or nonlinear system of equations
+  if(this->param.viscous_problem() or this->param.implicit_convective_problem())
   {
-    velocity_np = 0.0;
-    for(unsigned int i = 0; i < velocity.size(); ++i)
+    // Extrapolate old solutions to get a good initial estimate for the solver.
+    if(this->use_extrapolation)
     {
-      velocity_np.add(this->extra.get_beta(i), velocity[i]);
+      velocity_np = 0.0;
+      for(unsigned int i = 0; i < velocity.size(); ++i)
+      {
+        velocity_np.add(this->extra.get_beta(i), velocity[i]);
+      }
```

Comment:
So far, this extrapolation has also been done when solving the Euler equations with explicit convective term, i.e. even if there is no (non)linear system of equations to be solved in the momentum step. In that case, we only need to invert the mass matrix, for which we do not need, however, an extrapolation of the velocity as initial guess for an iterative solver.